nexus/db-queries/src/db/datastore/oximeter.rs

```diff
@@ -15,16 +15,32 @@ use crate::db::model::OximeterInfo;
 use crate::db::model::ProducerEndpoint;
 use crate::db::pagination::paginated;
 use crate::db::pagination::Paginator;
+use crate::db::queries;
 use async_bb8_diesel::AsyncRunQueryDsl;
 use chrono::DateTime;
 use chrono::Utc;
 use diesel::prelude::*;
+use diesel::result::DatabaseErrorKind;
+use diesel::result::Error as DieselError;
+use diesel::sql_types;
+use nexus_db_model::ProducerKindEnum;
 use omicron_common::api::external::DataPageParams;
 use omicron_common::api::external::Error;
 use omicron_common::api::external::ListResultVec;
 use omicron_common::api::external::ResourceType;
 use uuid::Uuid;
 
+/// Type returned when reassigning producers from a (presumably defunct)
+/// Oximeter collector.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum CollectorReassignment {
+    /// Success: `n` producers were reassigned to other collector(s).
+    Complete(usize),
+    /// Reassignment could not complete because there are no other collectors
+    /// available.
+    NoCollectorsAvailable,
+}
+
 impl DataStore {
     /// Lookup an oximeter instance by its ID.
```

Comment:
I might update this comment to be clear that it returns a non-deleted collector.